app/controllers/instance_user_role_requests_controller.rb

```diff
@@ -14,6 +20,21 @@ def create
     end
   end
 
+  def approve
+    @user_role_request.assign_attributes(user_role_request_params)
+
+    @success, instance_user = @user_role_request.approve_and_destroy!
+    if @success
+      flash.now[:success] = t('.success', user: instance_user.user.name, role: instance_user.role)
+    else
+      flash.now[:danger] = instance_user.errors.full_messages.to_sentence
```

Comment:
Perhaps we should put a user friendly error message here instead? Ask them to retry, or contact someone if the error persists.